File: frontend/src/api/page-layout.ts
import { request } from '@/utils/request'

// 页面布局接口
export interface PageLayout {
  id?: number
  layoutName: string
  layoutCode: string
  layoutType: string
  areaConfig?: string
  published?: boolean
  routePath?: string
  menuId?: number
  status?: boolean
  createBy?: string
  createTime?: string
  updateTime?: string
}

// 分页结果
export interface PageResult<T> {
  total: number
  records: T[]
  current: number
  size: number
  pages: number
}

// 区域配置接口
export interface AreaConfig {
  id: string
  title: string
  componentType: 'form' | 'table' | 'button'
  formId?: number
  tableId?: number
  buttonIds?: number[]
}

// 获取页面布局列表
export const getPageLayoutList = (params: any) => {
  return request.get<PageResult<PageLayout>>('/page-layout/list', params)
}

// 获取页面布局详情
export const getPageLayout = (id: number) => {
  return request.get<PageLayout>(`/page-layout/${id}`)
}

// 根据编码获取页面布局
export const getPageLayoutByCode = (layoutCode: string) => {
  return request.get<PageLayout>(`/page-layout/code/${layoutCode}`)
}

// 创建页面布局
export const createPageLayout = (data: PageLayout) => {
  return request.post<number>('/page-layout', data)
}

// 更新页面布局
export const updatePageLayout = (id: number, data: PageLayout) => {
  return request.put(`/page-layout/${id}`, data)
}

// 删除页面布局
export const deletePageLayout = (id: number) => {
  return request.delete(`/page-layout/${id}`)
}

// 发布页面布局
export const publishPageLayout = (id: number) => {
  return request.post(`/page-layout/${id}/publish`)
}

// 取消发布页面布局
export const unpublishPageLayout = (id: number) => {
  return request.post(`/page-layout/${id}/unpublish`)
}

File: backend/src/main/java/com/lowcode/entity/PageLayout.java
package com.lowcode.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.io.Serializable;
import java.time.LocalDateTime;
import java.util.List;

/**
 * 页面布局配置实体类
 */
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@TableName("page_layout")
public class PageLayout implements Serializable {

    private static final long serialVersionUID = 1L;

    @TableId(type = IdType.AUTO)
    private Long id;

    /**
     * 布局名称
     */
    private String layoutName;

    /**
     * 布局编码（唯一）
     */
    private String layoutCode;

    /**
     * 布局类型(top-bottom/left-right/tree-table/tabs)
     */
    private String layoutType;

    /**
     * 区域配置JSON
     * 格式示例：
     * {
     *   "areas": [
     *     {"id": "toolbar", "title": "工具栏", "componentType": "button"},
     *     {"id": "search", "title": "搜索区", "componentType": "form", "formId": 1},
     *     {"id": "content", "title": "内容区", "componentType": "table", "tableId": 1}
     *   ]
     * }
     */
    private String areaConfig;

    /**
     * 是否已发布
     */
    private Boolean published;

    /**
     * 路由路径
     */
    private String routePath;

    /**
     * 关联菜单ID
     */
    private Long menuId;

    /**
     * 状态（1=启用，0=禁用）
     */
    private Boolean status;

    /**
     * 是否删除
     */
    @TableLogic
    private Integer deleted;

    /**
     * 创建者
     */
    private String createBy;

    /**
     * 创建时间
     */
    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createTime;

    /**
     * 更新时间
     */
    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updateTime;

    /**
     * 关联的表单列表（非数据库字段）
     */
    @TableField(exist = false)
    private List<LowFormConfig> forms;

    /**
     * 关联的表格列表（非数据库字段）
     */
    @TableField(exist = false)
    private List<LowTableConfig> tables;
}
